src/csi/util.rs

```diff
@@ -190,14 +190,21 @@ pub fn spawn_grpc_task<R: Send + 'static>(
     sink: UnarySink<R>,
     task: impl Future<Output = DatenLordResult<R>> + Send + 'static,
 ) {
-    smol::spawn(async move {
-        let result = task.await;
-        match result {
-            Ok(resp) => async_success(sink, resp).await,
-            Err(e) => async_fail(sink, e).await,
-        }
-    })
-    .detach();
+    // spawn a thread to prevent the current thread from being blocked
+    // cannot use tokio spawn instead or there will be deadlock
+    thread::spawn(move || {
```

Comment:
Need to check why tokio::spawn() doesn't work here.

Reply:
done